adafruit_bno08x/__init__.py

```diff
@@ -482,32 +491,29 @@ class BNO08X:  # pylint: disable=too-many-instance-attributes, too-many-public-m
 
     """
 
-    def __init__(self, reset=None, debug=False):
-        self._debug = debug
-        self._reset = reset
+    def __init__(self, reset: Optional[dict] = None, debug: bool = False) -> None:
+        self._debug: bool = debug
+        self._reset: Optional[dict] = reset
         self._dbg("********** __init__ *************")
-        self._data_buffer = bytearray(DATA_BUFFER_SIZE)
-        self._command_buffer = bytearray(12)
-        self._packet_slices = []
+        self._data_buffer: bytearray = bytearray(DATA_BUFFER_SIZE)
+        self._command_buffer: bytearray = bytearray(12)
+        self._packet_slices: List[Any] = []
 
         # TODO: this is wrong there should be one per channel per direction
-        self._sequence_number = [0, 0, 0, 0, 0, 0]
-        self._two_ended_sequence_numbers = {
-            "send": {},  # holds the next seq number to send with the report id as a key
-            "receive": {},
-        }
-        self._dcd_saved_at = -1
-        self._me_calibration_started_at = -1
+        self._sequence_number: List[int] = [0, 0, 0, 0, 0, 0]
+        self._two_ended_sequence_numbers: Dict[int, int] = {}
+        self._dcd_saved_at: float = -1
+        self._me_calibration_started_at: float = -1
```

Comment:
I'm inclined to think these should either be `int` or have their value changed to `-1.0` if `float`s are valid. 

As-is the type declaration contradicts the value that is set in it.